app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -354,9 +317,13 @@ public Survey getSurveyMostRecentlyPublishedVersionByIdentifier(StudyIdentifier
             }
         }
         return new ArrayList<Survey>(map.values());
-        
     }
     
+    /**
+     * Get a specific survey version regardless of whether or not is has been deleted. This is the only call 
+     * that will return a deleted survey. With most scheduling now pointing to the most recently published 
+     * version (not a specific timestamped version), this method should be rarely called.
+     */
     @Override
     public Survey getSurvey(GuidCreatedOnVersionHolder keys) {
```

Comment:
Make sure when retrieving existing objects to work on them and then save, that you throw ENF if it comes back null.